File: DEMOS/DEMO-WebJS/DEMO-WebJS/ViewController.m
//
//  ViewController.m
//  DEMO-WebJS
//
//  Created by Alex D. on 3/10/16.
//  Copyright © 2016 ifnil. All rights reserved.
//

#import "ViewController.h"
//#import "UIWebViewController.h"
#import "WKWebViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)wkWebAction:(UIButton *)sender {

    WKWebViewController *vc = [[WKWebViewController alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}

- (IBAction)uiWebAction:(UIButton *)sender {
    //UIWebViewController *vc = [[UIWebViewController alloc] init];
    //[self.navigationController pushViewController:vc animated:YES];
}

@end

File: NaviTrans/NaviTrans/ViewController.swift
//
//  ViewController.swift
//  NaviTrans
//
//  Created by Songwen on 2018/8/30.
//  Copyright © 2018 Songwen. All rights reserved.


import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        let tab = UITabBarController()
        tab.title = "Tab"
        tab.viewControllers = [
            {
                let v = MyVC()
                v.title = "😄1"
                v.view.backgroundColor = .yellow
                return v
            }(),
            {
                let v = MyVC()
                v.title = "😄2"
                v.view.backgroundColor = .orange
                return v
            }(),
            {
                let v = MyVC()
                v.title = "😄3"
                v.view.backgroundColor = .red
                return v
            }()
        ]
        let navi = UINavigationController(rootViewController: tab)
        self.present(navi, animated: true, completion: nil)
    }
}

class MyVC: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = .orange

        let button = UIButton()
        button.frame = CGRect(x: 20, y: 100, width: 50, height: 30)
        button.setTitle("push", for: .normal)
        button.addTarget(self, action: #selector(pushButtonAction(sender:)), for: .touchUpInside)
        self.view.addSubview(button)
    }

    @objc func pushButtonAction(sender: UIButton) {
        let vc = MyVC()
        if let t = self.title {
            vc.title = t + "😄"
        }
        self.navigationController?.pushViewController(vc, animated: true)
    }
}

